lib/livebook/notebook/canvas_settings.ex

```diff
@@ -0,0 +1,34 @@
+defmodule Livebook.Notebook.CanvasSettings do
+  @moduledoc false
+
+  # Data structure representing a canvas of the notebook.
+
+  defstruct [:items]
+
+  alias Livebook.Utils
+  alias Livebook.Notebook.Cell
+
+  @type id :: Utils.id()
+
+  @type t :: %__MODULE__{
+          items: %{id() => item()}
+        }
+
+  @type item :: %{
+          x: non_neg_integer(),
+          y: non_neg_integer(),
+          w: non_neg_integer(),
+          h: non_neg_integer(),
+          outputs: list(Cell.indexed_output())
```

Comment:
We can't keep the outputs directly, because they wouldn't update on reevaluation and more importantly that's not something we could store in the .livemd and load later.

Because of reevaluation, it may be tricky to identify a specific output. We could store cell id + index, but if a random logger output pops up it can make the index point to a wrong output. I think we should treat all outputs of specific cell atomically in this context. In most cases each rich output would come from a different cell (or it should be easy to make it so). We can store canvas placement in each cell struct and persist in cell metadata (we don't have cell ids in .livemd, so we don't want to store references to that).

And one other note, I wonder what's the best way to store the placement. Coords + size seems natural, but that is not responsive and we will need scroll when the screen is smaller than the canvas. This is an issue for deployed apps where we want things to be more responsive. Perhaps we could make the x and width quantized, so it scales with the screen, but then height cannot be a fixed value either. I'm not sure what the best solution is, just pointing out.

@josevalim if you have concerns with any of the above let me know :)

Reply:
>  Coords + size seems natural, but that is not responsive and we will need scroll when the screen is smaller than the canvas.

Ah, I looked at gridstack and as far as I understand `x`, `y`, `w`, `h` are already quantized and it is responsive, so ignore that part :)

Reply:
unfortunately I will be on a short vacation till sunday, I will change this next week :)

Reply:
@jannikbecher there is absolutely no rush, thanks for working on this! Enjoy the vacation :D